modules/misunderstood/src/views/full/MainScreen.tsx

```diff
@@ -0,0 +1,300 @@
+          <td>Resolution</td>
+          <td>Updated</td>
+        </tr>
+      </thead>
+      <tbody>
+        {events.map((event, i) => (
+          <tr key={i}>
+            <td>{event.preview}</td>
+            <td>
+              {RESOLUTION[event.resolutionType]} {event.resolution}
+              {event.resolutionParams && (
+                <pre>
+                  <code>{JSON.stringify(event.resolutionParams, null, 2)}</code>
+                </pre>
+              )}
+            </td>
+            <td>{event.updatedAt}</td>
+          </tr>
+        ))}
+      </tbody>
+    </HTMLTable>
+  )
+
+const PendingList = ({
+  events,
+  totalEventsCount,
+  applyAllPending
+}: {
+  events: DbFlaggedEvent[]
+  totalEventsCount: number
+  applyAllPending: () => Promise<void>
+}) => (
+  <>
+    <h3>Pending Misunderstood ({totalEventsCount})</h3>
+    <div>
+      {events && events.length > 0 && (
+        <Button onClick={applyAllPending} intent={Intent.WARNING} icon="export" className="bp3-fill">
+          Apply all pending
+        </Button>
+      )}
+    </div>
+    <ResolvedEventsList events={events} />
+  </>
+)
+
+const AppliedList = ({ events, totalEventsCount }: { events: DbFlaggedEvent[]; totalEventsCount: number }) => (
+  <>
+    <h3>Applied Misunderstood ({totalEventsCount})</h3>
+    <ResolvedEventsList events={events} />
+  </>
+)
+
+const ChatPreview = ({ messages }: { messages: ContextMessage[] }) => (
+  <div className={style.chatPreview}>
+    {messages.map((message, i) => (
+      <div
+        key={i}
+        className={clsx(style.chatPreviewMessage, {
+          [style.chatPreviewMessage_Incoming]: message.direction === 'incoming',
+          [style.chatPreviewMessage_Outgoing]: message.direction === 'outgoing',
+          [style.chatPreviewMessage_Current]: message.isCurrent
+        })}
+      >
+        <div className={style.chatPreviewAvatar}>{message.direction === 'incoming' ? 'U' : 'B'}</div>
+        <div className={style.chatPreviewText}>{message.preview}</div>
+      </div>
+    ))}
+  </div>
+)
+
+const AmendForm = ({ mode, setMode }) => (
+  <div className={style.amendForm}>
+    <h5>What is this message type?</h5>
+
+    <ButtonGroup>
+      <Button
+        onClick={() => {
+          if (mode === RESOLUTION_TYPE.intent) {
+            return
+          }
+          setMode(RESOLUTION_TYPE.intent)
+        }}
+        intent={mode === RESOLUTION_TYPE.intent ? Intent.SUCCESS : Intent.NONE}
+      >
+        Goal
+      </Button>
+      <Button
+        onClick={() => {
+          if (mode === RESOLUTION_TYPE.qna) {
+            return
+          }
+          setMode(RESOLUTION_TYPE.qna)
+        }}
+        intent={mode === RESOLUTION_TYPE.qna ? Intent.SUCCESS : Intent.NONE}
+      >
+        Query
+      </Button>
+      {mode != null && (
+        <Button
+          onClick={() => {
+            setMode(null)
+          }}
+          icon="undo"
+        >
+          Undo
+        </Button>
```

Comment:
This could be a one liner with a small modification, ex: `{mode !== null && <Button onClick={() => setMode(null)} icon="undo" text="Undo" />}`